pkg/monitoring/vms/prometheus/prometheus.go

```diff
@@ -224,49 +224,99 @@ func (metrics *vmiMetrics) updateBlock(blkStats []stats.DomainStatsBlock) {
 			continue
 		}
 
-		if block.RdReqsSet || block.WrReqsSet {
-			desc := metrics.newPrometheusDesc(
-				"kubevirt_vmi_storage_iops_total",
-				"I/O operation performed.",
-				[]string{"drive", "type"},
+		blkLabels := []string{"drive"}
+		blkLabelValues := []string{block.Name}
+
+		if block.AliasSet {
+			blkLabelValues[0] = block.Alias
+		}
+
+		if block.RdReqsSet {
+			metrics.pushCustomMetric(
+				"kubevirt_vmi_storage_iops_read_total",
+				"I/O read operations",
+				prometheus.CounterValue,
+				float64(block.RdReqs),
+				blkLabels,
+				blkLabelValues,
 			)
+		}
 
-			if block.RdReqsSet {
-				metrics.pushPrometheusMetric(desc, prometheus.CounterValue, float64(block.RdReqs), []string{block.Name, "read"})
```

Comment:
I did not realize this before, but there is a good reason why `read` and `write` is only separated via labels. That is needed to properly create sums and other forms of aggregations. This must not be split into separate matrics. Why does this happen?

Reply:
@sradco ^^